### PR TITLE
API spec changes to support configuring workflow executors

### PR DESCRIPTION
**Pull request by nitishm**
Addresses #379

Signed-off-by: Nitish Malhotra <nitish.malhotra@gmail.com>

### CODE REVIEW

**Comment by nitishm** [comment]
@jonathan-innis we can merge this in rather than have you fork from my branch. Adding these API changes should be benign (I think). WDYT? That way you can just work off of main

File: api/v1alpha1/appgroup_types.go [hunk 1]
```diff
@@ -71,6 +71,46 @@ type ApplicationSpec struct {
...
+	KeptnExecutor ExecutorType = "keptn"
+	// CustomExecutor lets the user specify their own executor template to be used either as a deployment
+	// executor, a testing executor or for any other action to be taken on behalf of the application node.
+	CustomExecutor ExecutorType = "custom"
+)
+
+type Executor struct {
+	// DAG contains the dependency information
+	// +required
+	DAG `json:",inline"`
```
Comment (jonathan-innis) [inline api/v1alpha1/appgroup_types.go:102]: nit: newline here

File: api/v1alpha1/appgroup_types.go [hunk 2]
```diff
@@ -71,6 +71,46 @@ type ApplicationSpec struct {
...
+	// Type specifies the executor type to be run
+	// +required
+	Type ExecutorType `json:"type,omitempty"`
+
+	// Image allows the end user to specify the docker image name and tag
+	// to be executed by the workflow node
+	// +optional
+	Image string `json:"image,omitempty"`
```
Comment (jonathan-innis) [inline api/v1alpha1/appgroup_types.go:110]: nit: newline here

File: api/v1alpha1/appgroup_types.go [hunk 3]
```diff
@@ -71,6 +71,46 @@ type ApplicationSpec struct {
...
+	CustomExecutor ExecutorType = "custom"
+)
+
+type Executor struct {
+	// DAG contains the dependency information
+	// +required
+	DAG `json:",inline"`
+	// Type specifies the executor type to be run
+	// +required
+	Type ExecutorType `json:"type,omitempty"`
```
Comment (jonathan-innis) [inline api/v1alpha1/appgroup_types.go:105]: Can you specify the `kubebuilder:enum` annotation here?

File: api/v1alpha1/appgroup_types.go [hunk 4]
```diff
@@ -71,6 +71,46 @@ type ApplicationSpec struct {
...
+}
+
+// ExecutorType can either refer to a native executor (helmrelease and/or keptn) or
+// be a custom executor defined by the end-user
+type ExecutorType string
+
+const (
+	// HelmReleaseExecutor type is a "deployment" executor native to Orkestra that is responsible for
+	// deploying the application through the `HelmRelease` custom resource for the fluxcd helm-controller
+	HelmReleaseExecutor ExecutorType = "helmrelease"
```
Comment (jonathan-innis) [inline api/v1alpha1/appgroup_types.go:88]: nit: include newlines here

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

**Comment by jonathan-innis** [comment]
> @jonathan-innis we can merge this in rather than have you fork from my branch. Adding these API changes should be benign (I think). WDYT? That way you can just work off of main

Yes, in general, I agree that merging these changes is benign

**Review by jonathan-innis — COMMENTED** [review]
🚀

**Pull request merged**